piper/Dockerfile

```diff
@@ -12,14 +12,16 @@ ARG PIPER_RELEASE
 RUN \
     apt-get update \
     && apt-get install -y --no-install-recommends \
-        netcat \
+        netcat-traditional \
         python3 \
         python3-pip \
+        python3-venv \
     \
-    && pip3 install --no-cache-dir -U \
+    && python3 -m venv .venv \
```

Comment:
The Python in Debian bookworm gives this error when trying to upgrade system packages:

```
$ pip3 install --upgrade pip
error: externally-managed-environment

× This environment is externally managed
╰─> To install Python packages system-wide, try apt install
    python3-xyz, where xyz is the package you are trying to
    install.
    
    If you wish to install a non-Debian-packaged Python package,
    create a virtual environment using python3 -m venv path/to/venv.
    Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
    sure you have python3-full installed.
    
    If you wish to install a non-Debian packaged Python application,
    it may be easiest to use pipx install xyz, which will manage a
    virtual environment for you. Make sure you have pipx installed.
    
    See /usr/share/doc/python3.11/README.venv for more information.
```

There is likely a flag to override, but I figured it was best to go with the flow.

Reply:
I don't agree, as this is not a normal user runtime.

I'd suggest restoring the original and set the `PIP_BREAK_SYSTEM_PACKAGES=1` on the environment.

Reply:
Removed the venv